### DISCUSSION TITLE
404 error after deleting record

### DISCUSSION THREAD

**Discussion by Zach4531** [livewire/livewire#4270] [1. Help]
so we have a table that when you click on it, a modal pops up with the record information but some of that information uses livewire like our love/like button. Because of this I emit getRecordId everytime I open up the modal with new information (just sends component new id). Problem is when I delete a record on the page and then click on modal again it pops up with a 404 message. I want a way that can "Clean" out the deleted record. If it helps their is no recorded loaded on page load and it works fine.

`class Love extends Component
{

    public $record;

    public $recordId;

    public $sleep = false;

    public $listeners = [
        'getRecordId',
        'preventClick'
    ];

    public function mount($recordId = null)
    {
        if ($recordId) {
            $this->recordId = $recordId;
            $this->record= Record::find($this->recordId);
        }
    }

    public function getRecordId($recordId = null)
    {
        if ($recordId) {
            $this->recordId = $recordId;
            $this->record =Record::find($this->recordId);
        }
    }

    public function toggleLoved()
    {
        Auth::user()->lovedRecords()->toggle($this->record);
        if (Auth::user()->lovedRecords->contains($this->record)) {
            $this->record->author->notify(new NewRecordLove($this->record));
        }

        $this->sleep = true;
    }

    public function preventClick()
    {
        $this->sleep = false;
    }

    public function getIconProperty()
    {
        if ($this->record) {
            if (Auth::user()->lovedRecords->contains($this->record)) {
                return "fas fa-heart";
            }
            return "far fa-heart";
        }
        return "";
    }

    public function getCountProperty()
    {
        if ($this->record) {
            return $this->record->loved()->count();
        }
        return '?';
    }

    public function render()
    {
        return view('livewire.modal.love');
    }
}
`

**Comment by jonneroelofs** [comment]
To me the 404 suggests that the error is not coming from this component. Do you have a findOrFail or your table component perhaps?

**Reply by Zach4531** [reply]
I have a find but no findOrfail, so there actually 4 components that load that use the same initial record look up but when I look in console they show up erroring out or is that an effect from the problem you're saying

**Comment by joshhanley** [comment]
@Zach4531 the issue will probably be your `$record` and `$recordId` properties are still referencing the model that you deleted. So when another request happens Livewire tries to get that record from the database for you, but it no longer exists (hence the 404).

To fix it, make sure you set `$record` to null once it get's deleted from the database, so the component no longer has a reference to the eloquent model. You don't need to clear out `$recordId` as that is just an integer, but probably good practice to clear that too.

Hope this helps!
